### DISCUSSION TITLE
Best way to install a builder bob built module with its github URL?

### DISCUSSION THREAD

**Discussion by GNUGradyn** [callstack/react-native-builder-bob#612] [Q&A]
Hello. I've written a module that is incomplete. The android side is complete but the iOS side is not. This is a library that I need for my app but decided to open source. There is an iOS only alternative to my library, so I decided to put the iOS capability of my library on hold and just use the alternative for iOS and my library for android for my app for now. Because the package is usable on android but not complete in a way that prevents use on iOS, I want to add it to my app via its github URL as apposed to publishing it to NPM. 

Simply doing `yarn add username/repo` doesn't work, when I try to import it I get `Cannot find module '(library name)' or its corresponding type declarations.ts(2307)`. I am not certain but my guess is this is due to the fact lib is gitignored so all of the compiled sources including typescript annotations are missing. Simply pushing the lib folder and un-gitignoring it seems jank though, so I was wondering what is the "intended" way to accomplish this.

Thanks in advance!

**Comment by satya164** [comment]
The package needs to be built when installing from git - which can happen in `prepack` or `prepare` scripts. Unfortunately there's no single approach that works with all package managers.

By default, the library uses `prepare`, but you can change it to `prepack` in `package.json` and see if it works with your package manager. You can read more about which package managers support which method here https://callstack.github.io/react-native-builder-bob/build#manual-configuration

**Comment by DanielSRS** [comment]
@satya164 Sorry to ping you, but about that, I read the docs and also #497

Since package managers will ignore one or another, would be a terrible idea adding both? based on the docs, the only thing I can think is `bob build` running twice since `yarn publish, npm publish, pnpm publish` seems to execute both of them (prepare/prepack), but I not event sure if that is what happens. 

Sorry if its a obviously bad idea, I am not familiar with these scripts/publishing. Things like this have a lot on nuances which is not ease to find searching while not familiar with it

**Comment by satya164** [comment]
@DanielSRS it ultimately depends on you. i think as an alternative it might be possible to write a script that you use instead which can run the build and avoid running twice by detecting the `publish` command etc.

**Reply by DanielSRS** [reply]
That's awesome! thank you for responding in so short time. Also that seems a great alternative, never crossed my mind control this behaviour myself instead of trying not to break the package managers. This indeed sounds to be a more safer/controlled approach to considerate.
